packages/remix-dev/vite/static/refresh-utils.cjs

```diff
@@ -17,8 +17,9 @@ const enqueueUpdate = debounce(async () => {
 
     for (let route of routeUpdates.values()) {
       manifest.routes[route.id] = route;
-
-      let imported = await __hmr_import(route.url + "?t=" + Date.now());
+      let imported =
+        window.__remixRouteModuleUpdates.get(route.id) ??
+        (await __hmr_import(route.url + "?t=" + Date.now()));
```

Comment:
I think we should remove this and log a warning instead, otherwise this fallback will probably stick around indefinitely.